caffe2/serialize/intermediate_model.h

```diff
@@ -0,0 +1,318 @@
+#pragma once
+
+#include <unordered_map>
+
+#include <ATen/core/Allocator.h>
+
+#include "caffe2/proto/caffe2_pb.h"
+#include "caffe2/proto/torch_pb.h"
+#include "caffe2/serialize/inline_container.h"
+
+namespace at {
+
+struct IntermediateDeviceOption {
```

Comment:
first.. you use it only in Tensor. If you actually check which fields would be later used it's actually only deviceType + deviceId - the rest are the artifact of reusing DeviceOption (what does random seed for tensor mean?). So we can probably nuke them from this struct.

second.. it's less clear whether persisting the device info in the model is preferred or not. C2 right now has option keep_device (https://github.com/pytorch/pytorch/blob/master/caffe2/operators/load_save_op.h#L84) which is default false but sometime is true. PyTorch has the default of preserving the device and allowing to turn it off (https://discuss.pytorch.org/t/on-a-cpu-device-how-to-load-checkpoint-saved-on-gpu-device/349). But I guess that storing this info is useful